### PR TITLE
Natural sorting for chests

### PR DESCRIPTION
**Pull request by S-S-X**
Add natural sorting for chest items.

### CODE REVIEW

**Comment by S-S-X** [comment]
Have to fix this a bit, original unique items table is not unique enough for natural sorting by human readable name.

**Comment by S-S-X** [comment]
Fixed sorting of stacks that contain infotext metadata

File: technic_chests/inventory.lua [hunk 1]
```diff
@@ -67,6 +69,37 @@ function technic.chests.sort_inv(inv, mode)
 			end
 			return a.key < b.key
 		end)
+	elseif mode == 4 then
+		-- Natural
+		do -- Collect item stacks for sorting
```
Comment (OgelGames) [inline technic_chests/inventory.lua:74]: What does `do`, umm... do?

File: technic_chests/formspec.lua [hunk 2]
```diff
@@ -207,11 +209,8 @@ function technic.chests.get_receive_fields(data)
 			pipeworks.fs_helpers.on_receive_fields(pos, fields)
 		end
 		if data.sort and fields.sort_mode then
-			local value = meta:get_int("sort_mode") + 1
-			if value < 0 or value > 3 then
-				value = 0
-			end
-			meta:set_int("sort_mode", value)
+			local value = meta:get_int("sort_mode")
+			meta:set_int("sort_mode", (value + 1) % 5)
```
Comment (OgelGames) [inline technic_chests/formspec.lua:213]: Nice optimization there 👍

File: technic_chests/inventory.lua (hunk 1 above)
Comment (S-S-X) [reply technic_chests/inventory.lua:74]: It is just scope closure for scope control, same as { } in javascript or c/c++
So not required but added to separate collector and sort algorithm for readability

Comment (OgelGames) [reply technic_chests/inventory.lua:74]: Interesting...

File: technic_chests/inventory.lua [hunk 3]
```diff
@@ -67,6 +69,37 @@ function technic.chests.sort_inv(inv, mode)
...
+			local function name(stack)
+				return stack:get_meta():get("infotext")
+					or stack:get_description()
+					or stack:get_name()
+			end
+			local lookup = {}
+			for _,stack in pairs(list) do
+				local key = minetest.get_translated_string('', name(stack))
```
Comment (S-S-X) [inline technic_chests/inventory.lua:82]: Note that here if we had player data available sorting could use translated descriptions, empty locale seems to return default translation which is usually english.
Currently there's no player available when this function is called so players using other locale still gets items sorted by english descriptions.

**Comment by S-S-X** [comment]
Merging this, translation support can be easily added later. For now sorting will be done without translations.

**Comment by OgelGames** [comment]
I completely forgot about this PR :yum: I meant to test it and merge it sooner.

**Pull request merged**